include/libcgc.c

```diff
@@ -482,7 +483,35 @@ int deallocate(void *addr, cgc_size_t length) {
   return 0;
 }
 
-/* So this isn't really a random number generator. */
+
+cgc_prng *cgc_internal_prng = NULL;
+/**
+ * Initializes the prng for use with cgc_random and the secret page
+ */
+void try_init_prng() {
+    // Don't reinitialize
+    if (cgc_internal_prng != NULL) return;
+
+    // This will be hex encoded
+    const char *prng_seed_hex = getenv("seed");
+    if (prng_seed_hex == NULL) {
+        // TODO: Actually make this random
+        prng_seed_hex = "736565647365656473656564736565643031323334353637383961626364656600000000000000000000000000000000";
+    }
+
+    // Convert the hex encoded seed to a normal string
+    const char *pos = prng_seed_hex;
+    uint8_t prng_seed[BLOCK_SIZE * 3];
+    for(int i = 0; i < BLOCK_SIZE * 3; ++i) {
+        sscanf(pos, "%2hhx", &prng_seed[i]);
+        pos += 2;
+    }
+
+    // Create the prng
+    cgc_internal_prng = (cgc_prng *) malloc(sizeof(cgc_prng));
+    *cgc_internal_prng = cgc_init_prng(prng_seed);
```

Comment:
Check the length of the passed in env variable. Make sure it's `strlen() == ((BLOCKSIZE * 3) * 2)`